enhancements/78-dynamically-reclaiming-resources/README.md

```diff
@@ -0,0 +1,142 @@
+
+#### Job Parallelism Greater Than 1
+
+Whenever a Pod of a Job successfully completes its execution, Kueue will reclaim the resources that were associated with successful Pod. The Job might be still in `Running` state as there could be other Pods of the Job that are executing. Thus, Kueue will reclaim the resources of the succeeded Pods of the Job in an incremental manner until the Job completes in either `Succeeded` or `Failed` state. Whenever the Job completes, the only remaining owned resources of the Job will be reclaimed by Kueue.
+
+### Pod Failure Cases
+
+#### RestartPolicy of Pods as Never
+
+Whenever a Pod of a Job fails, the Job will recreate a Pod as replacement against the failed Pod. The Job will continue this process of creating new Pods until the Job reaches its `backoffLimit`(by default the backoffLimit is `6`). The newly created Pod against the failed Pod will reuse the resources of the failed Pod. Once the Job reaches its `backoffLimit` and does not have the required successful `.spec.completions` count, the Job is termed as a `Failed` Job. When the Job is marked as failed, no more Pods would be created by the Job. So, the remaining owned resources of the Job will be reclaimed by queue. 
+
+#### RestartPolicy of Pods as OnFailure
+
+When the Pods of the Job have `.spec.template.spec.restartPolicy = "OnFailure"`, the failed Pod stays on the node and the failed container within the Pod is restarted. The Pod might run indefinitely or get marked as `Failed` after the `.spec.activeDeadlineSeconds` is completed by Job if specified. In this case also, Kueue won't reclaim the resources of the Job until the Job gets completed as either `Failed` or `Succeeded`.
+
+Hence, as seen from the above discussed Pod failure cases, we conclude that the Pods of a Job which fail during its execution, its resources are not immediately reclaimed by Kueue. Only when the Job gets marks as `Failed`, the resources of failed Pods will be reclaimed by Kueue.
+
+## API
+
+A new field `ReclaimedPodSetPods` is added to the `.status` of Workload API.
+
+```go
+// WorkloadStatus defines the observed state of Workload
+type WorkloadStatus struct {
+    // conditions hold the latest available observations of the Workload
+    // current state.
+    // +optional
+    // +listType=map
+    // +listMapKey=type
+    Conditions []WorkloadCondition `json:"conditions,omitempty"`
+
+    // ReclaimedPodSetPods are the pods (by PodSet name) which have
+    //completed their execution successfully and resources of the pods
+	//are reclaimed.
+    // +optional
+    ReclaimedPodSetPods ReclaimedPodSetPods `json:"reclaimedPodSetPods"`
+}
+
+type ReclaimedPodSetPods map[string]struct{
+    Count int
+    Pods []string
```

Comment:
why do we need the pod names? That could be a lot of data.

Reply:
Pod names was kept to record the pods whose resources are reclaimed and avoid reclaiming again. But I agree, that could be lot pods.

Reply:
The job controller API already counts the number of finished pods. Why not use that? https://kubernetes.io/docs/concepts/workloads/controllers/job/#job-tracking-with-finalizers